src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -478,24 +484,24 @@ private void emitGlobalArraySetter(String holderClass, IndexList indexList,
         incrAlign();
         emitDocComment(varTree, docHeader);
         if (Utils.isStructOrUnion(elemType)) {
-            appendLines(STR."""
-                public static void \{javaName}(\{indexList.decl()}, MemorySegment varValue) {
-                    MemorySegment.copy(varValue, 0L, \{javaName}(\{indexList.use()}), 0L, \{layoutString(elemType)}.byteSize());
-                }
-                """);
+            appendLines(MessageFormat.format("""
+        public static void {0}({1}, MemorySegment varValue) '{'
+            MemorySegment.copy(varValue, 0L, {0}({2}), 0L, {3}.byteSize());
+        '}'
+        """, javaName, indexList.decl(), indexList.use(), layoutString(elemType)));
```

Comment:
Is this the only place where MessageFormat is used? Why just here?

(also, indentation was dropped here)